src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -0,0 +1,380 @@
+                }
+
+                string methodName = _reader.GetAttribute("name");
+                if (!String.IsNullOrEmpty(methodName))
+                {
+                    bool foundMatch = false;
+                    foreach (MethodDesc method in type.GetMethods())
+                    {
+                        if (method.Name == methodName)
+                        {
+                            foundMatch = true;
+                            ProcessMethod(method);
+                        }
+                    }
+
+                    if (!foundMatch)
+                    {
+                        //Context.LogWarning($"Could not find method '{name}' on type '{type.GetDisplayName()}'", 2009, _xmlDocumentLocation);
+                    }
+                }
+            }
+        }
+
+        protected virtual void ProcessMethod(MethodDesc method)
+        {
+        }
+
+        private void ProcessField(TypeDesc type)
+        {
+            if (ShouldProcessElement())
+            {
+                string fieldName = _reader.GetAttribute("name");
+                if (!String.IsNullOrEmpty(fieldName))
+                {
+                    FieldDesc field = type.GetField(fieldName);
+
+                    if (field == null)
+                    {
+                        //Context.LogWarning($"Could not find field '{name}' on type '{type.GetDisplayName()}'", 2012, _xmlDocumentLocation);
+                    }
+                    else
+                    {
+                        ProcessField(field);
+                    }
+                }
+            }
+        }
+
+        protected virtual void ProcessField(FieldDesc field)
+        {
+        }
+
+        protected MethodDesc GetMethod(TypeDesc type, string signature)
+        {
+            foreach (MethodDesc meth in type.GetMethods())
+                if (signature == GetMethodSignature(meth, false))
+                    return meth;
+
+            return null;
+        }
+
+        public static string GetMethodSignature(MethodDesc meth, bool includeGenericParameters)
+        {
+            StringBuilder sb = new StringBuilder();
+            CecilTypeNameFormatter.Instance.AppendName(sb, meth.Signature.ReturnType);
+            sb.Append(' ');
+            sb.Append(meth.Name);
+            if (includeGenericParameters && meth.HasInstantiation)
+            {
+                sb.Append('`');
+                sb.Append(meth.Instantiation.Length);
+            }
+
+            sb.Append('(');
+            for (int i = 0; i < meth.Signature.Length; i++)
+            {
+                if (i > 0)
+                    sb.Append(',');
+
+                CecilTypeNameFormatter.Instance.AppendName(sb, meth.Signature[i]);
+            }
+
+            sb.Append(')');
+            return sb.ToString();
+        }
+
+        private bool ShouldProcessElement()
+        {
+            return true;
+
+            /*string feature = _reader.GetAttribute("feature");
```

Comment:
Note to self: uncomment - I just want to enable all feature switches so that it does things.